python/ray/autoscaler/aws/example-cloudwatch.yaml

```
@@ -0,0 +1,85 @@
+# An unique identifier for the head node and workers of this cluster.
+cluster_name: cloudwatch
+
+# Cloud-provider specific configuration.
+provider:
+    type: aws
+    region: us-west-2
+    availability_zone: us-west-2a
+    # Start by referencing Unified CloudWatch Agent json config file path here.
+    # Note that this config assumes Unified CloudWatch Agent is installed on every cluster node.
+    # For AMI with Unified CloudWatch Agent pre-installed, see https://github.com/amzn/amazon-ray.
+    cloudwatch:
+        # For updates to existing Unified CloudWatch Agent json config file:
+        # Config file is uploaded to the AWS Systems Manager (SSM) Parameter Store during `ray up`.
+
+        # Note only parameters with "AmazonCloudWatch-" prefix are allowed for access with CloudWatchAgentAdminPolicy.
+        # See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/create-iam-roles-for-cloudwatch-agent.html for additional details.
+        # The following SSM Parameter Store Config Keys are used to store a remote cache of the last config applied:
+        # Unified CloudWatch Agent config SSM Parameter name: AmazonCloudWatch-ray_agent_config_{cluster_name}
+        # Every time you run `ray up` to update your cluster, we compare your local CloudWatch config file contents
+        # to the SSM Parameter Store's contents for that config and, if they differ, then the associated CloudWatch
+        # config will be applied and uploaded to the SSM Parameter Store.
+        # Note: For CloudWatch json config files, variables referencing is supported.
+        # Currently we recursively find and replace three custom variables: {instance_id}, {region}, {cluster_name}.
+        agent:
+            # Path to Unified CloudWatch Agent config file
+            # The Unified CloudWatch Agent config file is a JSON file with three sections: agent, metrics and logs.
+            # See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-Configuration-File-Details.html for additional details.
+            # Note in this example-cloudwatch-agent-config.json file, the following log files are shipped to CloudWatch:
+            # 1. All log files under /tmp/ray/session_*/logs/**.out will be in CloudWatch/Logs {cluster_name}-ray_logs_out log group.
+            # 2. All log files under /tmp/ray/session_*/logs/**.err will be in CloudWatch/Logs {cluster_name}-ray_logs_err log group.
+            # 3. All Prometheus metrics will be under CloudWatch/Metrics/ray-prometheus namespace.
+            # 4. log stream name will be the head node instance id in this example.
```

```suggestion
            # 4. CloudWatch Log Stream names will be the same as your cluster head node's EC2 instance ID.
```
Also, are we sure this is true? Are worker node instance IDs never recorded? I thought I had seen worker node instance IDs recorded as Log Stream names when using this feature in the past.